### ISSUE TITLE
Introduce display_name for threat.indicator

### ISSUE THREAD

**Issue by maxcold** [elastic/ecs#1998]
**Summary**

**Motivation**:

**Detailed Design**:

Provide additional details around the design of the proposed changes.

* Field names
* Example values for the fields
* Suggested appropriate datatypes
* Any example events that map to the proposed use case(s)

**Comment by maxcold** [comment]
@jamiehynds fyi, following up on our discussion

**Comment by peasead** [comment]
I think that `threat.indicator.display_name` is not a bad idea - I would think that you'd replicate the atomic indicator as the value.

Also, here's the `email` ECS fieldset https://github.com/elastic/ecs/blob/main/schemas/email.yml, which could be nested under `threat.indicator.`

**Comment by maxcold** [comment]
Thanks for the comments, I agree that `threat.indicator.name` is probably more consistent with the rest of the schema. Also agree that it probably make sense to nest the `email` fieldset, as currently only `threat.indicator.email.address` exists to my understanding at it doesn't match with https://github.com/elastic/ecs/blob/main/schemas/email.yml, (`threat.indicator.email.address` should be `threat.indicator.email.from.address` if I understand correctly when following the existing `email` schema)

**Comment by peasead** [comment]
Yep, you're following the schema properly!

**Back story**
We made some decisions early on regarding the directionality of some types of indicators when writing the `threat.*` ECS fieldset.

We felt that ECS fieldsets that incorporate directionality (`source.ip`, `destination.domain`, `email.from.address`, etc.) would lead to confusion when trying to get indicators into the right fields. As an example, if the indicator is `1.2.3.4` - is it command & control infrastructure or is it the source of a password spraying campaign. Should it be `threat.indicator.source.ip : 1.2.3.4` or `threat.indicator.destination.ip : 1.2.3.4`? In the email example, is this the source of a phishing email or the reply-to address?

Contextually, it could be both - so if threat provider A marked it as a source and threat provider B marked it as a destination; you could have duplicate threat indicator matches or worse, contextually incorrect assumptions based on how someone viewed the directionality.

We opted to avoid the confusion of directionality by not including it and doing `threat.indicator.ip : 1.2.3.4`, `threat.indicator.domain`,  `threat.indicator.email.address`, etc and allowing an analyst to determine //what// happened using other fields populated during the enrichment - like `network.direction : ingress|egress`

**Commentary**
I think if we wanted to reapproach directionality, we could do that, but having looked at the feed data over time, I think directionality would be difficult.

**Comment by maxcold** [comment]
@peasead Thanks a lot for the Back Story, it answers a lot of questions that I had as I wasn't really thinking about the directionality of the indicators. It makes total sense! So I think it should stay the way it is now, meaning for `email-addr` IoC type as just `threat.indicator.email.address`. 
But now I'm not sure what part of the description you were commenting with this note
```
Also, here's the email ECS fieldset https://github.com/elastic/ecs/blob/main/schemas/email.yml, which could be nested under threat.indicator.
```
Can you clarify so we are on the same page?

**Comment by peasead** [comment]
I think you can disregard that. I didn't fully grasp what you were asking until I started the larger response.

Sorry!

**Comment by jamiehynds** [comment]
@maxcold based on the discussions above with @peasead, do you think `threat.indicator.name` is a more suitable fit than the original `threat.indicator.display_name` proposal? 

Could you also provide a proposal for the field description and allowed values, which we'd include in the ECS documentation? As an example, here's the description and allowed values for an upcoming addition to `event.category` - https://github.com/elastic/ecs/issues/2028#issuecomment-1216805476

@ebeahan given that this proposal is a relatively minor change, I'm assuming an RFC isn't warranted, but pinging you just incase you feel otherwise.

**Comment by djptek** [comment]
@maxcold is it likely that e.g. `threat.indicator.name` could have **multiple values for distinct indicators within a single event?**

If, so, we'd perhaps want to consider adding

`threat.enrichments.indicator.name` 

where `enrichments` is an array containing multiple indicators as well as

`threat.indicator.name`

**Comment by maxcold** [comment]
@jamiehynds yes, I think it makes sense to have `threat.indicator.name` for consistency with the rest of ECS
Description: "The display name of the Indicator of Compromise in UI friendly format"
Allowed values: there is a mapping between the type of IoC and which field from the document should be used for the `name` field. How should I go about allowed values in this case?
You mentioned that the change is minor but we not only want to introduce this new field, but also want it to be populated for TI integrations based on the mapping logic I added into description. Is it in the scope of this issue or I will need to add a new issue to implement the logic in the Integrations?

@djptek good point, I think it makes sense to add `threat.enrichments.indicator.name` too in addition to `threat.indicator.name` which follows the same logic and share the same Description/Allowed Values and serves the IoC name when added as enrichment to Alerts for example

**Comment by djptek** [comment]
You can specify expected values in the schema `yml` for a field, see e.g. [`event.category`](https://github.com/elastic/ecs/blob/9a3611bc818228c3e91e76094ccf0ffe4fe14867/schemas/event.yml#L166)

**Comment by maxcold** [comment]
@djptek thanks for providing an example. One question, you linked to `expected_event_types` attribute, did you mean to link to `allowed_values`? I'm just not sure how `expected_event_types` is relevant here.
As for `allowed_values` the problem is that the proposed `threat.indicator.name` is not a enum, the value and the type of this value depends on the `threat.indicator.type`, here are some examples
For an `'ipv4-addr' indicator
```
{
  threat:
  {
    indicator: {
      _id: '123',
      type: 'ipv4-addr'
      ip: '1.1.1.1'
    }
  }
}
```
"to be" state with the new field populated
```
{
  threat:
  {
    indicator: {
      _id: '123',
      name: '1.1.1.1'
      type: 'ipv4-addr'
      ip: '1.1.1.1'
    }
  }
}
```

for a file type indicator
```
{
  threat:
  {
    indicator: {
      _id: '123',
      type: 'file'
      file: { hash: {'md5': 'md5_hash', sha256: 'sha256_hash'}}
    }
  }
}
```

"to be" state with the new field populated
```
{
  threat:
  {
    indicator: {
      _id: '123', 
      name: 'sha256_hash'
      type: 'file'
      file: { hash: {'md5': 'md5_hash', sha256: 'sha256_hash'}}
    }
  }
}
```

**Comment by djptek** [comment]
Hi @maxcold sorry, I gave completely the wrong example there. I intended to give an example for `expected_values`

**expected_values (optional): An array of expected values for the field. Schema consumers can validate integrations and mapped data against the listed values. These values are the recommended convention, but users may also use other values.**

From your examples:
- `name: 'sha256_hash'`
- `name: '1.1.1.1'`

I think `expected_values` might be a good fit

**Comment by maxcold** [comment]
@djptek no worries, I didn't know about `expected_values`, thanks for bring it up! We can definitely provide some example values in the expected_values, but after looking at how `expected_values` is currently used https://github.com/elastic/ecs/search?q=expected_values I'm not sure if it is a good fit. It seems like it describes the exact values that can appear in a field so that the consumers can build validation for it. In our case, we can only provide example values, which might confuse the schema consumers. But I might be completely wrong about `expected_values`, happy to provide more example fields if needed

**Comment by djptek** [comment]
@maxcold `expected_values` would work if you can provide examples, they doesn't need to be exhaustive and are not intended for validation

This is not the same as `allowed_values`, which should be validated always

**Comment by maxcold** [comment]
@djptek got it, thanks!
Here is the list of example values for `threat.indicato.name` field
```
5.2.75.227
2a02:cf40:add:4002:91f2:a9b2:e09a:6fc6
https://example.com/some/path
example.com
373d34874d7bc89fd4cefa6272ee80bf
b0e914d1bbe19433cc9df64ea1ca07fe77f7b150b511b786e46e007941a62bd7
email@example.com
HKLM\\SOFTWARE\\Microsoft\\Active
13335
00:00:5e:00:53:af
8008
```

@jamiehynds anything else I can do to help moving it forward?

**Comment by maxcold** [comment]
hi @jamiehynds , as we are coming closer to the 8.6 release cycle anything we can help with to get this on the roadmap for the release?

**Comment by maxcold** [comment]
@ebeahan @jamiehynds @djptek hey folks, what can our team do to help move this forward?

**Comment by ebeahan** [comment]
@maxcold sorry for missing the ping before.

Is this summary still the direction we're taking: https://github.com/elastic/ecs/issues/1998#issuecomment-1225737413? 

If so, the next steps would be for someone on your team to open a PR with the requested changes, and we'll review and discuss further as needed.

**Comment by maxcold** [comment]
@ebeahan yes, that's still the idea. I will add an issue of creating a PR to ECS schema to our backlog then. What about changes in the ti_* integrations? It would be good if these integration populate the field automatically, how do we make this happen?

**Comment by maxcold** [comment]
@ebeahan btw what is usually the approach for backfilling the data if a new ECS field is introduced (or if there is another change in the schema)? Is there a common way to handle such cases? Specifically for `threat.indicator.name` it would be good to provide a way for users to add this field to the already existing data. Is it a good idea in general? how should we handle this?

**Comment by ebeahan** [comment]
@maxcold there's not a common approach I'm familiar with.

Typically when a field is introduced in ECS, the change is made to the data source or integration to start populating that field. New events and indices will have the field, but existing data will not.

**Comment by maxcold** [comment]
got it, thanks!
